sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -467,10 +492,158 @@ The `node` class provides the {crs}[common reference semantics].
+passed to the executable graph's update function.
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+Parameters:
+
+* `executionRange` - The new value for the Range.
+
+Exceptions:
+
+* Throws with error code `invalid` if the type of the node is not a kernel
+  execution.
+
+|===
+
+==== Dynamic Parameters
+
+Dynamic parameters are arguments to a node's command-group which can be updated
+by the user after the node has been added to a graph. Updating the value of a
+dynamic parameter will be reflected in the modifiable graph which contains this
+node. These updated nodes can then be passed to an executable graph to update
+it with new values.
+
+The type of the underlying object a dynamic parameter represents is set at
+compile time using a template parameter. This underlying type can be an
+accessor, a pointer to a USM allocation, scalar passed by value, or a raw byte
+representation of the argument. The raw byte representation is intended to
+enable updating arguments set using
+link:../proposed/sycl_ext_oneapi_raw_kernel_arg.asciidoc[sycl_ext_oneapi_raw_kernel_arg].
+
+Dynamic parameters are registered with nodes in a modifiable graph, with each
+registration associating one or more node arguments to the dynamic parameter
+instance. Registration happens inside the command-group that the node represents
+, and is done using the argument index (known when argument originally set using
+`set_arg()`/`set_args()`). It is valid for a node argument to be registered with
+more than one dynamic parameter instance.
+
+Registration is done by using the argument index (known when argument
+originally set using `set_arg()`/`set_args()`).
+
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+The `dynamic_parameter` class provides the {crs}[common reference semantics].
+
+Table {counter: tableNumber}. Member functions of the `dynamic_parameter` class.
+[cols="2a,a"]
+|===
+|Member Function|Description
+
+|
+[source,c++]
+----
+dynamic_parameter(command_graph<graph_state::modifiable> graph, const ValueT &param);
+----
+|Constructs a dynamic parameter object that can be registered with command graph
+nodes. The param argument is used only to deduce the template parameter ValueT,
+which is the type of the kernel argument this `dynamic_parameter` represents.
+
+Parameters:
+
+* `graph` - Graph containing the nodes which will have dynamic parameters.
+* `param` - Value used for CTAD.
+
+|
+[source,c++]
+----
+void register_with_node(const handler &cgh, int argIndex);
+----
+|Callable only from command-group scope when adding a node to a graph
+using the explicit API. Associate the dynamic parameter
+with the kernel parameter of the node that encapsulates this command-group.
+Parameters are associated by their index which was given when they were
+registered with `handler`.
```

Comment:
We should somehow say here that this function is only useful when the kernel is defined as a free function kernel.

Reply:
I've added some wording to this effect, though specifically to only use it when args are set explicitly using indices, such as with `handler::set_arg()` since this leaves compatibility with anything (present or future) which uses `set_arg`.